services/src/layers/external.rs

```diff
@@ -22,11 +24,9 @@ use geoengine_operators::source::{GdalLoadingInfo, OgrSourceDataset};
 use serde::{Deserialize, Serialize};
 
 #[async_trait]
-pub trait DataProviderDefinition:
-    CloneableDataProviderDefinition + Send + Sync + std::fmt::Debug
-{
+pub trait DataProviderDefinition<D: GeoEngineDb>: Send + Sync + std::fmt::Debug {
     /// create the actual provider for data listing and access
-    async fn initialize(self: Box<Self>) -> Result<Box<dyn DataProvider>>;
+    async fn initialize(self: Box<Self>, db: D) -> Result<Box<dyn DataProvider>>;
```

Comment:
Why do we need access to a `GeoEngineDb` for initializing?

Reply:
The `DatasetLayerListingProvider` needs access to the Datasets. That's why I had to change the `DataProvider` trait. Furthermore, we always wanted to have credentials per user and data provider. To store and access those the `DataProvider` also needs access to the `GeoEngineDb`. That's why I think this change makes sense.